### ISSUE TITLE
Support the ability to see the current activity.

### ISSUE THREAD

**Issue by RokeJulianLockhart** [MuntashirAkon/AppManager#1464]
### Please check before submitting an issue

- [X] I am using the latest version of App Manager
- [X] I have searched the issues and haven't found anything relevant
- [X] I have read the docs

### Describe a description of the new feature

I request the ability to see the activity of the currently visible application, because it can be incredibly difficult to locate a specific activity from the package-specific search functionality alone.

### Describe the solution you'd like

Provide a dynamic notification, accessibility overlay, and/or separate window (useable with relevant multi-window multitasking option in *Developer options* enabled). The 1st and 3rd options utilize entirely native functionality already employed by App Manager.

### Describe alternatives you've considered

Due to the inconvenience of https://android.stackexchange.com/questions/185720/how-to-get-current-foreground-activity-name/185725#comment351982_185756, and the fact that both https://android.stackexchange.com/questions/185720/how-to-get-current-foreground-activity-name/185725#comment351983_185725 and https://github.com/Asura-64/Current-Activity/issues/1#issue-2633031318 don't work, I'm rather at a loss for *any* alternative.

### Additional context

As aforementioned at [XDA Forums](https://xdaforums.com/t/discussion-muntashirakon-app-manager.4674128/#post-89791591:~:text=I'm%20looking%20for%20different%20functionality,if%20it%20really%20isn't%20present.):

> I'm looking for different functionality that I rather expected to find in this application - a counterpart to the functionality offered by [https://github.com/codehasan/Curren...982e282f3e6a3a5d523a/README.md#--how-it-works](https://github.com/codehasan/Current-Activity/blob/0cbd59b6f94182e32c1c982e282f3e6a3a5d523a/README.md#--how-it-works), now that that application is defunct. I'll file an FR if it really isn't present.

**Comment by MuntashirAkon** [comment]
> I request the ability to see the activity of the currently visible application, because it can be incredibly difficult to locate a specific activity from the package-specific search functionality alone.

> Provide a dynamic notification, accessibility overlay, and/or separate window (useable with relevant multi-window multitasking option in _Developer options_ enabled). The 1st and 3rd options utilize entirely native functionality already employed by App Manager.

We already have **UI tracker** feature in v3.x.x which you can utilize to figure out the current app. However, I haven't really done any investigation on displaying the current activity. Let's see if there's any API that allows that (if there are any, I believe it will require privileged access).

I don't plan on adding a notification system, because it will make things more complicated (e.g., you'll always get the notification window/System UI to be current package when you view your notification, so a mechanism has to be made to skip checking for **System UI**, and so on, which will have other consequences as well since the System UI package scope is not limited to just displaying notifications).

**Comment by RokeJulianLockhart** [comment]
> I haven't really done any investigation on displaying the current activity. Let's see if there's any API that allows that (if there are any, I believe it will require privileged access).

@MuntashirAkon, that would be wonderful! The method by which I can acquire the current activity isn't particularly important, as long as there is *a* method.

However, note that the referenced application did manage to implement both a notification and overlay, so if the code is reusable, there's a reference implementation.

**Comment by MuntashirAkon** [comment]
> However, note that [the referenced application](https://github.com/codehasan/Current-Activity/compare/main...Asura-64:Current-Activity:main#diff-7fa6aef292187a049f7a4d6060d8df3ba212d838789c78940bd363344b1c38cdR9) did manage to implement both a notification and overlay, so if the code is reusable, there's a reference implementation.

They simply [ignored all the system class names](https://github.com/codehasan/Current-Activity/blob/main/app/src/main/java/io/github/ratul/topactivity/service/AccessibilityMonitoringService.java#L68-L69) which is neither correct nor useful in this case.

BTW, you can get the current activity using the following command (tested in Android 14 and later):

```sh
dumpsys activity activities | grep topResumedActivity= | awk '{ print $3 }'
```

**Comment by RokeJulianLockhart** [comment]
> https://github.com/MuntashirAkon/AppManager/issues/1464#issuecomment-2458269229

@MuntashirAkon, that's perfect! Thank you. Might I ask where you got that from? I'm amazed I wasn't able to find it.

**Comment by MuntashirAkon** [comment]
> Might I ask where you got that from? I'm amazed I wasn't able to find it. I've posted

It's in the Android source code.

**Comment by MuntashirAkon** [comment]
Implemented in 45ff4f8f3f236fc16362232cb82ec353cee6e3e1

**Comment by MuntashirAkon** [comment]
Those plain windows, they aren't associated with any activities.
